lib/ecto/changeset.ex

```diff
@@ -2723,11 +2727,12 @@ defmodule Ecto.Changeset do
     unique_constraint(changeset, [field], opts)
   end
 
-  def unique_constraint(changeset, [first_field | _] = fields, opts) do
+  def unique_constraint(changeset, fields, opts) when is_list(fields) do
```

Comment:
I would actually revert this to what it was because we need to pattern match on the list not being empty anyway!

```suggestion
  def unique_constraint(changeset, [first_field | _] = fields, opts) do
```

Reply:
Good catch!